#! /bin/sh -e

# DP: Look for gcc-3.3 as gcc with Ada support.

dir=
if [ $# -eq 3 -a "$2" = '-d' ]; then
    pdir="-d $3"
    dir="$3/"
elif [ $# -ne 1 ]; then
    echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
    exit 1
fi
case "$1" in
    -patch)
        patch $pdir -f --no-backup-if-mismatch -p0 < $0
        cd ${dir}gcc && autoconf
        ;;
    -unpatch)
        patch $pdir -f --no-backup-if-mismatch -R -p0 < $0
        rm ${dir}gcc/configure
        ;;
    *)
        echo >&2 "`basename $0`: script expects -patch|-unpatch as argument"
        exit 1
esac
exit 0


--- gcc/aclocal.m4~	Sat Mar  2 00:37:42 2002
+++ gcc/aclocal.m4	Sat Apr  6 13:01:36 2002
@@ -312,6 +312,7 @@
 	    ${ac_tool_prefix}$user_cc	$user_cc	\
 	    ${ac_tool_prefix}gcc	gcc		\
 	    ${ac_tool_prefix}cc		cc		\
+	    ${ac_tool_prefix}gcc-3.3	gcc-3.3		\
 	    ${ac_tool_prefix}gnatgcc	gnatgcc		\
 	    ${ac_tool_prefix}gnatcc	gnatcc		\
 	    ${ac_tool_prefix}adagcc	adagcc		\
